packages/openapi-code-generator/src/typescript/server/typescript-express/typescript-express-router-builder.ts

```diff
@@ -0,0 +1,243 @@
+        .add("parseRequestInput", "responseValidationFactory")
+    } else if (this.schemaBuilder instanceof JoiBuilder) {
+      this.imports
+        .from("@nahkies/typescript-express-runtime/joi")
+        .add("parseRequestInput", "responseValidationFactory")
+    }
+  }
+
+  protected buildOperation(builder: ServerOperationBuilder): string {
+    const statements: string[] = []
+
+    const symbols = this.operationSymbols(builder.operationId)
+    const params = builder.parameters(symbols)
+
+    if (params.path.schema) {
+      statements.push(constStatement(symbols.paramSchema, params.path.schema))
+    }
+    if (params.query.schema) {
+      statements.push(constStatement(symbols.querySchema, params.query.schema))
+    }
+    if (params.header.schema) {
+      statements.push(
+        constStatement(symbols.requestHeaderSchema, params.header.schema),
+      )
+    }
+    if (params.body.schema) {
+      statements.push(
+        constStatement(symbols.requestBodySchema, params.body.schema),
+      )
+    }
+
+    const responder = builder.responder(
+      "ExpressRuntimeResponder",
+      "ExpressRuntimeResponse",
+    )
+
+    this.operationTypes.push({
+      operationId: builder.operationId,
+      statements: [
+        buildExport({
+          name: symbols.responderName,
+          value: responder.type,
+          kind: "type",
+        }),
+        buildExport({
+          name: symbols.implTypeName,
+          value: `(
+                    params: ${params.type},
+                    respond: ${symbols.responderName},
+                    req: Request,
+                    res: Response,
+                  ) => Promise<ExpressRuntimeResponse<unknown>>`,
+          kind: "type",
+        }),
+      ],
+    })
+
+    statements.push(`
+const ${symbols.responseBodyValidator} = ${builder.responseValidator()}
+
+// ${builder.operationId}
+router.${builder.method.toLowerCase()}(\`${route(builder.route)}\`, async (req: Request, res: Response, next: NextFunction) => {
+  try {
+   const input = {
+    params: ${params.path.schema ? `parseRequestInput(${symbols.paramSchema}, req.params, RequestInputType.RouteParam)` : "undefined"},
+    query: ${params.query.schema ? `parseRequestInput(${symbols.querySchema}, req.query, RequestInputType.QueryString)` : "undefined"},
+    body: ${params.body.schema ? `parseRequestInput(${symbols.requestBodySchema}, req.body, RequestInputType.RequestBody)` : "undefined"},
+    headers: ${params.header.schema ? `parseRequestInput(${symbols.requestHeaderSchema}, req.headers, RequestInputType.RequestHeader)` : "undefined"}
+   }
+
+   const responder = ${responder.implementation}
+
+   const response = await implementation.${symbols.implPropName}(input, responder, req, res)
+    .catch(err => { throw ExpressRuntimeError.HandlerError(err) })
+
+   const { status, body } = response instanceof ExpressRuntimeResponse ? response.unpack() : response
+
+    res.status(status)
+
+    if (body !== undefined) {
```

Comment:
Should there be a way for the user to cause `next` to be called even after a response has been sent?

Eg: pass our own `next` function to the implementation that records if its called, allowing us to call the real `next` conditionally here after the promise resolves